### ISSUE TITLE
[Question] Jasper/Quartznet ASR - Recommended length per file is between 5 and 25 seconds. Why?

### ISSUE THREAD

**Issue by ShantanuNair** [NVIDIA/NeMo#1459]
**Describe your question**

Recommended length per file is between 5 and 25 seconds for most models. I'm curious where/how we arrive at this limitation, and what this means for results I get from running model.transcribe() on a audio longer than 25s. Some insight into where/why we have this limitation would be appreciated.

**Environment overview (please complete the following information)**

 - Environment location: [Cloud(Collab)]
 - Method of NeMo install: [from git with branch == v1.0.0b2]

**Comment by rbracco** [comment]
Not an expert, but I'll try to help (assuming you're talking about ASR). I believe the recommendation is due to most of the training data being in that range. Since the ASR models, Quartznet and Jasper, are fully convolutional, there's no RNN and I believe this means longer than 25s shouldn't cause too much of a problem, but give it a shot and see what kind of word error rate/accuracy you get.

The potential problems I see with...
**Clips < 5 seconds** 
Not having enough context may cause problems. The receptive field of the convolutional ASR models is at least a few seconds, so really short bursts may cause a lot of the input to the model to be zeros (padding) resulting in weird results. Not sure if this actually happens

**Clips > 25s**
I would guess this isn't a problem for cpu inference. On GPU (inference, and especially training) you'd be limited to very small batch sizes and that could potentially cause problems. 

Hope that helps and holds you over until a more experienced contributors is able to answer your question.

**Comment by ShantanuNair** [comment]
@rbracco Thank you!! That's what I thought regarding a lack of RNN meaning >25s shouldn't be a problem. I ran with it and it seems okay, but still giving rather unusable output. I haven't run any issues with batch size since I'm testing out transcriptions with a single audio file using the pretrained QuartzNet15x5Base-En and haven't done any additional training yet. 

```
oh i meat five researchers yesrano it's crazy so that's what i' saying it depends on what you consider my team um my research team within nhat product there's two of us that are like um full on workers and then we have some lab assistants that help out ofon the brigger product
```

Seems like beam-searching via a Language Model would help but I feel the quality is still lacking even without an LM. 

@okuchaiev Any comments regarding the original question?

**Comment by titu1994** [comment]
A bit late, but better late than never. The above comment is correct, Clips over 20 seconds of length would exhaust GPU memory, forcing tiny batch sizes and reducing training efficiency. Clips of any duration less than that is fine, but if its too short, then either the model wont have enough context to emit tokens.
